packages/atclient/include/atclient/atclient.h

```diff
@@ -110,14 +114,15 @@ int atclient_get_selfkey(const atclient *atclient, atclient_atkey *atkey, char *
  * associated with your value.
  *
  * @param atclient the atclient context (must satisfy the two conditions stated above)
+ * @param root_conn initialized root connection
  * @param atkey the populated atkey to get the value from (must satisfy the two conditions stated above)
  * @param value the buffer to hold value gotten from atServer
  * @param valuelen the buffer length allocated for the value
  * @param valueolen the output length of the value gotten from atServer
  * @return int 0 on success
  */
-int atclient_get_publickey(const atclient *atclient, const atclient_atkey *atkey, char *value, const size_t valuelen,
-                           size_t *valueolen);
+int atclient_get_publickey(atclient *atclient, atclient_connection *root_conn, const atclient_atkey *atkey, char *value,
+                           const size_t valuelen, size_t *valueolen);
```

Comment:
Is it true that you can't use the connections inside of atclient if it is passed in as a const pointer? I've been doing `get_selfkey` and `get_publickey` stuff just fine.

I thought that const pointer means it is pass by reference and you are not allowed to change the values inside (you can't say something like: `atclient->secondary_connection = some_new_connection()`)

Reply:
We shouldn't denote it as const because mbedtls_ssl_write requires a writeable ssl context.